models/instrument.py

```diff
@@ -0,0 +1,610 @@
+                    Instrument.name == inst.name,
+                    Instrument.telescope == inst.telescope,
+                )
+            ).first()  # should be unique combination of name/telescope
+
+            if db_inst is None:
+                session.add(inst)
+                session.commit()
+            else:
+                if db_inst != inst:
+                    print(f'Updating instrument {inst.name} on DB!')
+                    db_inst.update(inst)  # also make sure sections are updated
+                    session.commit()
+
+
+class DemoInstrument(Instrument):
+
+    __mapper_args__ = {
+        "polymorphic_identity": "DemoInstrument",
+    }
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.name = 'DemoInstrument'
+        self.telescope = 'DemoTelescope'
+        self.aperture = 2.0
+        self.focal_ratio = 5.0
+        self.square_degree_fov = 0.5
+        self.pixel_scale = 0.41
+        self.read_time = 2.0
+        self.read_noise = 1.5
+        self.dark_current = 0.1
+        self.gain = 2.0
+        self.non_linearity_limit = 10000.0
+        self.saturation_limit = 50000.0
+        self.allowed_filters = ["g", "r", "i", "z", "Y"]
+
+        for k, v in kwargs.items():
+            setattr(self, k, v)
+
+    @staticmethod
+    def _make_sections():
+        """
+        Make a single section for the DEMO instrument.
+
+        Returns
+        -------
+        sections: list of SensorSection
+            The sections for this instrument.
+        """
+        return [SensorSection(0, 512, 1024)]
+
+    def load_section(self, filename, section_id):
+        """
+        A spoof load method for this demo instrument.
+        The data is just a random array.
+        The instrument only has one section,
+        so the input must be 0.
+
+        Parameters
+        ----------
+        filename: str
+            The filename of the exposure file.
+            In this case the filename is not used.
+        section_id: str or int
+            The identifier of the SensorSection object.
+            This instrument only has one section, so this must be 0 or "0".
+
+        Returns
+        -------
+        data: np.ndarray
+            The data from the exposure file.
+        """
+        section = self.get_section_by_identifier(section_id)
+
+        return np.random.poisson(10, (section.size_y, section.size_x))
+
+    def read_header(self, filename):
+        return {}
+
+    def get_filename_regex(self):
+        return [r'Demo']
+
+
+class DECam(Instrument):
+    __mapper_args__ = {
+        "polymorphic_identity": "DECam",
+    }
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.name = 'DECam'
+        self.telescope = 'Blanco'
+        self.aperture = 4.0
+        self.focal_ratio = 2.7
+        self.square_degree_fov = 3.0
+        self.pixel_scale = 0.2637
+        self.read_time = 20.0
+        self.read_noise = 7.0
```

Comment:
OK so parsing header keys is moved to the Instrument class. You can use the base class parsing or override/enhance the functions that do that. 
There are two categories of keywords to look for: with/without default. E.g., "mjd" has no default, and if the header doesn't have that value the Exposure object will be left with `mjd=None` (we can check that value later on). 
The other category will use the instrument defaults if the key is missing in the file. E.g., the "gain" will just use the instrument's default value if not given in the header.

Reply:
The only question is: do we want to save the "parsed" header values in a separate dictionary on each Exposure? Do we even want to save the entire fits header into the DB? maybe we should only save the "important" keywords, after parsing them into common names (e.g., `exp_time` instead of `EXPTIME`) and leave the full header only on the file on disk?

Reply:
In previous pipelines, I've saved the entire header (as a JSONB) to the database, but that does seem a little excessive.  The only reason really to do that is if you'd want to pull out a table of values that includes some header keywords, and I *have* done that, but rarely.

If we're just going to save the important keywords, then we already have database fields for those things, so I don't think there's a value in saving that additionally in a header field.